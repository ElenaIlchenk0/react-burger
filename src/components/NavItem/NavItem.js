import React from 'react';
import navItemStyles from './NavItem.module.css';
import PropTypes from 'prop-types';

class NavItem extends React.Component {

    render() {
        return (
            <div onClick={() => this.props.onActiveNavItemId(this.props.id)}
                className={`${this.props.isActive ? navItemStyles.active : navItemStyles.inactive} ${navItemStyles.item} pt-4 pb-4 pr-5 pl-5`}>
                {this.props.icon}
                <div>{this.props.title}</div>
            </div>

        )
    }

}

NavItem.propTypes = {
    id: PropTypes.number,
    icon: PropTypes.any.isRequired,
    title: PropTypes.string.isRequired
}

export default NavItem;