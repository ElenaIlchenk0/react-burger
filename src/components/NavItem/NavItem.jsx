import React from 'react';
import navItemStyles from './NavItem.module.css';
import PropTypes from 'prop-types';

class NavItem extends React.Component {

    

    render() {
        return (
            <div className={`${this.props.isActive ? navItemStyles.active : navItemStyles.inactive} ${navItemStyles.item} pt-4 pb-4 pr-5 pl-5`}>
                <a href="#" className={navItemStyles.link}>
                    {this.props.icon}
                    <div>{this.props.title}</div>
                </a>
            </div>
        )
    }

}

NavItem.propTypes = {
    isActive: PropTypes.bool.isRequired,
    id: PropTypes.number,
    icon: PropTypes.object.isRequired,
    title: PropTypes.string.isRequired
}

export default NavItem;