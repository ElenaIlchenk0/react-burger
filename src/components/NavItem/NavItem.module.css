.item {
    cursor: pointer;
}

.active {
    opacity: 1;
}

.inactive {
    opacity: .6;
}

.link {
    display: flex;
    gap: 8px;
    align-items: center;
    text-decoration: none;
    color: inherit;
}