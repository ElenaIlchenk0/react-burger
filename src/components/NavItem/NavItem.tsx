import React, { ReactElement } from 'react';
import navItemStyles from './NavItem.module.css';
import { Link } from 'react-router-dom';

interface INavItem {
    isActive: boolean,
    title: string,
    icon: ReactElement,
}

const NavItem: React.FC<INavItem> = ({ isActive, title, icon }) => {
    let link = '';
    switch (title) {
        case 'Конструктор':
            link = '/'
            break;

        case 'Лента заказов':
            link = '/'
            break;

        case 'Личный кабинет':
            link = '/profile'
            break;

        default:
            link = '/'
            break;
    }
    
    return (
        <div className={`${isActive ? navItemStyles.active : navItemStyles.inactive} ${navItemStyles.item} pt-4 pb-4 pr-5 pl-5`}>
            <Link to={link} className={navItemStyles.link}>
                {icon}
                <div>{title}</div>
            </Link>
        </div>
    )

}

export default NavItem;
