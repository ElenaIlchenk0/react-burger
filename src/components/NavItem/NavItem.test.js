import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import NavItem from './NavItem';
import { BurgerIcon, ProfileIcon } from '@ya.praktikum/react-developer-burger-ui-components';
import { BrowserRouter } from 'react-router-dom';


const navItemProps = {
    isActive: true,
    icon: <ProfileIcon type="primary" />,
    title: 'Личный кабинет'
}

describe('NavItem in AppHeader', () => {

    it('renders without errors', () => {
        const tree = renderer
            .create(
                <BrowserRouter>
                    <NavItem {...navItemProps} />
                </BrowserRouter>
                )
        expect(tree).toMatchSnapshot();
    }); 

    it('should has correct title', () => {
        render(
            <BrowserRouter>
                <NavItem {...navItemProps} />
            </BrowserRouter>
        );
        expect(screen.getByTestId('title')).toHaveTextContent('Личный кабинет');
    }); 

    it('should has correct link', () => {
        render(
            <BrowserRouter>
                <NavItem {...navItemProps} />
            </BrowserRouter>
        );
        expect(screen.getByTestId('link')).toHaveAttribute('href', '/profile')
    }); 
    
    it('should receive class "active" when "isActive: true" and not receive "inactive"', () => {
        render(
            <BrowserRouter>
                <NavItem {...navItemProps} />
            </BrowserRouter>
        );
        const wrapper = screen.getByTestId('wrapper')

        expect(wrapper.classList.contains('active')).toBe(true);
        expect(wrapper.classList.contains('inactive')).not.toBe(true);
    });

    it('should receive class "inactive" when "isActive: false" and not receive "active"', () => {
        const navItemProps = {
            isActive: false,
            icon: <BurgerIcon type="primary" />,
            title: 'Конструктор'
        }

        render(
            <BrowserRouter>
                <NavItem {...navItemProps} />
            </BrowserRouter>
        );
        const wrapper = screen.getByTestId('wrapper')

        expect(wrapper).toHaveClass('inactive');
        expect(wrapper).not.toHaveClass('active');;
    });

})
