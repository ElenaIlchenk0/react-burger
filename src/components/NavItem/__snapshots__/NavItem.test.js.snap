// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem in AppHeader renders without errors 1`] = `
<div
  className="active item pt-4 pb-4 pr-5 pl-5"
  data-testid="wrapper"
>
  <a
    className="link"
    data-testid="link"
    href="/profile"
    onClick={[Function]}
  >
    <svg
      fill="#F2F2F3"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M17.068 6.56875C17.068 10.0837 14.8959 13.1 12 13.1C9.10401 13.1 6.93203 10.0837 6.93203 6.56875C6.93203 3.05385 8.80118 1 12 1C15.1988 1 17.068 3.05385 17.068 6.56875ZM3.10524 20.9572C3.53926 21.4607 5.40556 23 12 23C18.5944 23 20.4608 21.4607 20.8947 20.9572C20.9792 20.8593 21.0103 20.7362 20.9971 20.6088C20.8969 19.638 20.0015 15.3 12 15.3C3.99848 15.3 3.10311 19.638 3.00291 20.6088C2.98975 20.7362 3.02077 20.8593 3.10524 20.9572Z"
        fillRule="evenodd"
      />
    </svg>
    <div
      data-testid="title"
    >
      Личный кабинет
    </div>
  </a>
</div>
`;
